src/main/java/org/mybatis/spring/annotation/MapperScannerRegistrar.java

```diff
@@ -145,6 +157,48 @@ void registerBeanDefinitions(AnnotationMetadata annoMeta, AnnotationAttributes a
 
   }
 
+  private List<TypeFilter> typeFiltersFor(AnnotationAttributes filterAttributes) {
+
+    List<TypeFilter> typeFilters = new ArrayList<>();
+    FilterType filterType = filterAttributes.getEnum("type");
+
+    for (Class<?> filterClass : filterAttributes.getClassArray("value")) {
+      switch (filterType) {
+        case ANNOTATION:
+          Assert.isAssignable(Annotation.class, filterClass,
+            "An error occured when processing a @ComponentScan " + "ANNOTATION type filter: ");
+          @SuppressWarnings("unchecked")
+          Class<Annotation> annoClass = (Class<Annotation>) filterClass;
+          typeFilters.add(new AnnotationTypeFilter(annoClass));
+          break;
+        case ASSIGNABLE_TYPE:
+          typeFilters.add(new AssignableTypeFilter(filterClass));
+          break;
+        case CUSTOM:
+          Assert.isAssignable(TypeFilter.class, filterClass,
+            "An error occured when processing a @ComponentScan " + "CUSTOM type filter: ");
+          typeFilters.add(BeanUtils.instantiateClass(filterClass, TypeFilter.class));
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown filter type " + filterType);
```

Comment:
I prefer such as following message.

* `Cannot specify the 'value' or 'type' attribute if use the '{filterType}' in exclude filter of @MapperScan.`